src/automata/StreamRequester.cpp

```diff
@@ -7,6 +7,42 @@
 
 namespace reactivesocket {
 
+void StreamRequesterBase::onNext(Payload request) {
+  switch (state_) {
+    case State::NEW: {
+      state_ = State::REQUESTED;
+      // FIXME: find a root cause of this assymetry; the problem here is that
+      // the Base::request might be delivered after the whole thing is shut
+      // down, if one uses InlineConnection.
+      size_t initialN = initialResponseAllowance_.drainWithLimit(
+          Frame_REQUEST_N::kMaxRequestN);
+      size_t remainingN = initialResponseAllowance_.drain();
+      // Send as much as possible with the initial request.
+      CHECK_GE(Frame_REQUEST_N::kMaxRequestN, initialN);
+      auto flags = initialN > 0 ? FrameFlags_REQN_PRESENT : FrameFlags_EMPTY;
+      Frame_REQUEST_STREAM frame(
```

Comment:
I think this change is a great improvement. I would suggest taking one step further in this direction.
Instead of virtual onNext method which is identical for both StreamRequesterBase and SubscriptionRequesterBase, except for the frame type
   Frame_REQUEST_STREAM frame(
vs
   Frame_REQUEST_SUB frame(

you can create a new virtual method called sendRequestFrame which will be called from the onNext method and the different implementations will be:
SubscriptionRequesterBase::sendRequestFrame(...) {
       Frame_REQUEST_SUB frame(
           streamId_,
           flags,
           static_cast<uint32_t>(initialN),
           FrameMetadata::empty(),
           std::move(request));
       connection_->onNextFrame(frame);
}

StreamRequesterBase::sendRequestFrame(...) {
       Frame_REQUEST_STREAM frame(
           streamId_,
           flags,
           static_cast<uint32_t>(initialN),
           FrameMetadata::empty(),
           std::move(request));
       connection_->onNextFrame(frame);
}